src/redux/features/documentProducer/variables/variablesSelector.ts

```diff
@@ -226,6 +226,25 @@ const associateDeliverableVariableValues = (
   };
 };
 
+export const selectAllVariablesWithValues = createCachedSelector(
+  (state: RootState, requestId: string, projectId: number, maxValueId?: number) =>
+    state.documentProducerAllVariables[requestId],
+  (state: RootState, requestId: string, projectId: number, maxValueId?: number) =>
+    state.documentProducerVariableValuesList[variableListCompositeKeyFn({ projectId, maxValueId })],
+  (variableList, valueList) => {
+    if (variableList?.data && valueList?.data) {
+      const variables = variableList.data;
+      const values = valueList.data;
+
+      return variableList.data.map((v: Variable) => associateNonSectionVariableValues(v, values, variables));
```

Comment:
I'm getting an error here on staging. Maybe variableList?.data is empty? Should validate variableList?.data.lenght > 0 ?

Reply:
Hmm that is strange, because we are checking that `variableList.data` is truthy on line 235. That means that it is something other than an empty array. Can you tell what `variableList.data` is before it fails?